odoo/fields.py

```diff
@@ -1601,14 +1601,20 @@ def __init__(self, string=Default, **kwargs):
             kwargs['translate'] = bool(kwargs['translate'])
         super(_String, self).__init__(string=string, **kwargs)
 
-    def _setup_attrs(self, model_class, name):
-        super()._setup_attrs(model_class, name)
-        if self.prefetch is None:
-            # do not prefetch complex translated fields by default
-            self.prefetch = not callable(self.translate)
-
     _related_translate = property(attrgetter('translate'))
 
+    def setup(self, model):
+        setup_done = self._setup_done
+        super().setup(model)
+        # Force the _rec_name to be prefetch
+        if setup_done or self.prefetch is not None:
+            return
+        # Do not prefetch translated fields by default
+        # expect if it is the _rec_name of the model (in `_setup_base` of models.py)
+        self.prefetch = not (self.translate or callable(self.translate))
+        if model._rec_name == self.name:
+            self.prefetch = True
```

Comment:
The method `setup` is not supposed to be overridden, but `setup_related` and `setup_nonrelated` are.
```suggestion
def setup_nonrelated(self, model):
    super().setup_nonrelated(model)
    if self.prefetch is None:
        # translated fields are not prefetched by default except for _rec_name
        self.prefetch = not self.translate or model._rec_name == self.name

def setup_related(self, model):
    super().setup_related(model)
    if self.prefetch is None:
        # translated fields are not prefetched by default except for _rec_name
        self.prefetch = not self.translate or model._rec_name == self.name
```

Reply:
Your version seems cleaner indeed.